pkg/gcptarget/gcptarget.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Name is the target name for GCPTarget
+const Name = "validation.gcp.forsetisecurity.org"
+
+// GCPTarget is the constraint framework target for FCV
+type GCPTarget struct {
+}
+
+var _ client.TargetHandler = &GCPTarget{}
+
+// New returns a new GCPTarget
+func New() *GCPTarget {
+	return &GCPTarget{}
+}
+
+// MatchSchema implements client.MatchSchemaProvider
+func (g *GCPTarget) MatchSchema() apiextensions.JSONSchemaProps {
+	return apiextensions.JSONSchemaProps{
+		Properties: map[string]apiextensions.JSONSchemaProps{
+			"target": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+			"exclude": {
+				Type: "array",
+				Items: &apiextensions.JSONSchemaPropsOrArray{
+					Schema: &apiextensions.JSONSchemaProps{
+						Type: "string",
+					},
+				},
+			},
+		},
+	}
+}
+
+// GetName implements client.TargetHandler
+func (g *GCPTarget) GetName() string {
+	// Done
+	return Name
+}
+
+// Library implements client.TargetHandler
+func (g *GCPTarget) Library() *template.Template {
+	// TODO - lib impl
+	return libraryTemplate
+}
+
+// ProcessData implements client.TargetHandler
+func (g *GCPTarget) ProcessData(obj interface{}) (bool, string, interface{}, error) {
+	return false, "", nil, errors.Errorf("Storing data for referential constraint eval is not supported at this time.")
+}
+
+// HandleReview implements client.TargetHandler
+func (g *GCPTarget) HandleReview(obj interface{}) (bool, interface{}, error) {
+	switch asset := obj.(type) {
+	case *validator.Asset:
+		return g.handleAsset(asset)
+	case map[string]interface{}:
+		return true, asset, nil
+	}
+
+	return false, nil, nil
+}
+
+// handleAsset handles input from FCV assets as received via the gRPC interface.
+func (g *GCPTarget) handleAsset(asset *validator.Asset) (bool, interface{}, error) {
+	if asset.Resource == nil {
+		return false, nil, errors.Errorf("forseti asset's resource field is nil %s", asset)
+	}
+	asset2.CleanStructValue(asset.Resource.Data)
+	m := &jsonpb.Marshaler{
+		OrigName: true,
+	}
+	var buf bytes.Buffer
+	if err := m.Marshal(&buf, asset); err != nil {
+		return false, nil, errors.Wrapf(err, "marshalling to json with asset %s: %v", asset.Name, asset)
+	}
+	var f interface{}
+	err := json.Unmarshal(buf.Bytes(), &f)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "marshalling from json with asset %s: %v", asset.Name, asset)
+	}
+	return true, f, nil
+}
+
+// HandleViolation implements client.TargetHandler
+func (g *GCPTarget) HandleViolation(result *types.Result) error {
+	// Done? - this mutates the result after it's provided, we may have to add metadata
```

Comment:
You may want to populate result.Resource{}, though this is not strictly necessary if you always know the input resource.

Reply:
We're always going to be passing the input for now.  I've added a TODO for doing this once Audit gets implemented.

Reply:
This is going to be identical to result.Review, now populated.